### DISCUSSION TITLE
Crosh api + Rigtools ?

### DISCUSSION THREAD

**Discussion by ghost** [3kh0/ext-remover#1309] [Help JOIN TITANIUM NETWORK TO SHARE PLEASE] [answered]
could be use the rigtools evaluated code function  to than run and use chrome.terminalPrivate some how to create a crosh instance ???

**Comment by schoolexploitkid** [comment, marked as answer]
the extension has to have terminal permissions

**Reply by ghost** [reply]
Oh ok

**Reply by akabutnicer** [reply]
in addition, I do have a working way to modify any extension's manifest (credit to writable); therefore allowing me to change the extension key to, say, secure shell's. however, all I need is a way to fully reload (or power wash) the extension to load the new manifest

**Reply by s0urce-c0de** [reply]
> in addition, I do have a working way to modify any extension's manifest (credit to writable); therefore allowing me to change the extension key to, say, secure shell's. however, all I need is a way to fully reload (or power wash) the extension to load the new manifest

method plz. you can try flipping the allow access to file urls switch.

**Reply by akabutnicer** [reply]
really? thank you so much I never knew this!

**Reply by s0urce-c0de** [reply]
> really? thank you so much I never knew this!

what is the method tho...

**Reply by akabutnicer** [reply]
its currently redacted, can't say much until we find a working restarting process

**Reply by FunnyCorgi** [reply]
@akabutnicer why not have it use the pre-existing chrome.management toggle extensions payload to restart the extension(the default)?

**Reply by FunnyCorgi** [reply]
like modify it, make it work for that extension id, put it into a file in another extension(you will have to check if the extension has chrome.management permissions), open it, close it, delete it, all after modifying the extension.
you could also make it put it in a different extension then the modified ext-replaced one(call the exploit ext-modifier, and make a non-skid version you modify https://vscode.dev to make it work with the extension where the extension acts like a github repo and you can "commit" to it and modify and create and edit files under the extensions filesystem).

**Reply by akabutnicer** [reply]
> @akabutnicer why not have it use the pre-existing chrome.management toggle extensions payload to restart the extension(the default)?

i love this one, I'll try it.

**Reply by akabutnicer** [reply]
i can just use the 'assessment assistant' extension to carry on with this idea

**Reply by FunnyCorgi** [reply]
@akabutnicer why not just use allow the user to put in an extension id(because not everyone may have that extension)?

**Reply by FunnyCorgi** [reply]
You could also just make a wrapper for https://vscode.dev so it allows non-skids to directly edit the extension's filesystem. You could have seperate payloads for a skid version to replace with specific extensions.
You could also add a way in the non-skid version to instead give the manifest of the targeted extension proper perms of an extension you want to put in a directory of the targeted extension. Doing this could allow an extension to be combine multiple other extensions into another extension(modify the manifest as well to have options for each extensions popup(when you click the extension's icon in the toolbar) and to have stuff in that like:
- `Default function`
- `<extra extension one>`
- `<extra extension two>`
- `etc.`

**Reply by FunnyCorgi** [reply]
We don't need a wrapper! it looks like its all here: https://github.com/microsoft/vscode/

**Comment by FunnyCorgi** [comment]
@akabutnicer can you release to me a prototype of this? I need to see how you got access to writing to the extension root because webkit doesn't seem to provide that in the rigtools concept as far as i am aware.
At least tell me how you access the root of the extension's filesystem, I am working on figuring out how to get vscode into the extension.

**Reply by akabutnicer** [reply]
adding vscode to an extension is retarded behavior, but anyways hit me up on discord, `@akabutnice`

**Reply by FunnyCorgi** [reply]
@akabutnicer will do tonight or tomorrow, can't right now.
